swarm/network/simulation/simulation.go

```diff
@@ -85,7 +85,10 @@ func New(services map[string]ServiceFunc) (s *Simulation) {
 		name, serviceFunc := name, serviceFunc
 		s.serviceNames = append(s.serviceNames, name)
 		adapterServices[name] = func(ctx *adapters.ServiceContext) (node.Service, error) {
-			b := new(sync.Map)
+			b, ok := s.buckets[ctx.Config.ID]
```

Comment:
Notice the `s.mu.Lock()` just a few lines down there. Here you access a map in closure, that access was only made before after acquiring the mutex. Is that possible you get a race here?